docs/adr/010_device_request_queue.md

```diff
@@ -0,0 +1,164 @@
+# Device Request Queue
+
+- **Feature**: [#1480]
+- **Author(s)**: Atif Aziz
+- **Status**: Proposed
+
+## Context
+
+LoRa devices, which can reach multiple LNS, pose an interesting challenge for
+Edge Hub since Edge Hub wasn't designed for devices competing for gateways.
+When a device connection is made to IoT Hub, Edge Hub continues to maintain
+that connection in the open state (for network resiliency) until explicitly
+closed. Moreover, IoT Hub allows maintining only a single connection per
+device. When another is opened, the existing one closed. If a device can reach
+multiple LNS then Edge Hub and IoT Hub begin an aggressive ping-pong of
+connections being opened and closed. In other words, as each LNS or Edge Hub
+tries to open a connection for the same device, IoT Hub closes the connection
+for the other. But then the other tries to re-open the connection and the
+ping-pong continues. This problem needs addressing to avoid scalability
+limitations.
+
+The ping-pong can be pevented by electing a leader LNS for a device and
+announcing the decision to others using a C2D message. An LNS receiving such a
+message will then close that connection so that Edge Hub does't continue to
+maintain the connection in the open state through retires. The handling of the
+message needs to know:
```

Comment:
```suggestion
The ping-pong can be prevented by electing a leader LNS for a device through
the function race determination and as a fallback announcing the decision to 
others using a C2D message. An LNS receiving such a message will then close 
that connection so that Edge Hub doesn't continue to maintain the connection 
in the open state through retires. The handling of the message needs to know:
```

Reply:
@atifaziz - btw, this is unfortunately a good example that the line length limitation does not really work with determining the diff easily. If you need to insert something and have to adjust the alignment, things will end up
being the same.